From a5085a858a557f6fe40169d0d072395c37c347b8 Mon Sep 17 00:00:00 2001
From: Aaditya Kumar <aaditya.kumar.30@gmail.com>
Date: Sat, 22 Dec 2012 05:13:40 +0530
Subject: [PATCH 084/859] AXFS: fix build failure due to undefined symbol
 kern_path_parent()

This patch fixes following build failure when axfs is compiled
as a module. The build failure is caused by
Commit-ID: a7e526129f6ea298513ef4ce7aa6db08fc069d14

ERROR: "kern_path_parent" [fs/axfs/axfs.ko] undefined!

Signed-off-by: Aaditya Kumar <aaditya.kumar@ap.sony.com>
---
 fs/axfs/axfs_bdev.c  | 18 ------------------
 fs/axfs/axfs_super.c | 10 ++++++----
 2 files changed, 6 insertions(+), 22 deletions(-)

diff --git a/fs/axfs/axfs_bdev.c b/fs/axfs/axfs_bdev.c
index 0083a72..3c176fe 100644
--- a/fs/axfs/axfs_bdev.c
+++ b/fs/axfs/axfs_bdev.c
@@ -94,24 +94,6 @@ void axfs_copy_block(struct super_block *sb, void *dst_addr, u64 fsoffset,
 	}
 }
 
-int axfs_is_dev_bdev(char *path)
-{
-	struct nameidata nd;
-	int ret = false;
-
-	if (!path)
-		return false;
-	nd.flags |= LOOKUP_FOLLOW;
-	if (kern_path_parent(path, &nd))
-		return false;
-
-	if (S_ISBLK(nd.path.dentry->d_inode->i_mode))
-		ret = true;
-
-	path_put(&nd.path);
-	return ret;
-}
-
 int axfs_verify_bdev_sizes(struct super_block *sb, int *err)
 {
 	u64 io_dev_size;
diff --git a/fs/axfs/axfs_super.c b/fs/axfs/axfs_super.c
index e29288f..a912437 100644
--- a/fs/axfs/axfs_super.c
+++ b/fs/axfs/axfs_super.c
@@ -754,6 +754,7 @@ axfs_get_sb_address(struct file_system_type *fs_type,
 {
 	int mtdnr;
 	char *sd = sbi->second_dev;
+	struct dentry *dp = NULL;
 
 	if (sbi->phys_start_addr == 0)
 		return false;
@@ -767,11 +768,12 @@ axfs_get_sb_address(struct file_system_type *fs_type,
 
 	if (axfs_is_dev_mtd(sd, &mtdnr))
 		return axfs_get_sb_mtd(fs_type, flags, sd, sbi);
-	else if (axfs_is_dev_bdev(sd))
-		return axfs_get_sb_bdev(fs_type, flags, sd, sbi);
-	else
-		return mount_nodev(fs_type, flags, sbi, axfs_fill_super);
 
+	dp = axfs_get_sb_bdev(fs_type, flags, sd, sbi);
+	if (!IS_ERR_OR_NULL(dp))
+		return dp;
+
+	return mount_nodev(fs_type, flags, sbi, axfs_fill_super);
 }
 
 /* helpers for parse_axfs_options */
-- 
1.8.1.4

